integ-tests/src/test/java/com/google/integration/ColumnsTest.java

```diff
@@ -0,0 +1,91 @@
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.generic.GenericRecord;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ColumnsTest extends TestBase {
+
+  private static Connection connection;
+
+  @BeforeClass
+  public static void beforeClass() throws SQLException {
+    connection = DriverManager.getConnection(URL_DB, USERNAME_DB, PASSWORD_DB);
+  }
+
+  @Test
+  public void columnsTest() throws SQLException, IOException {
+    final String sqlPath = "src/main/java/com/google/sql/columns.sql";
+    final String avroFilePath = ET_OUTPUT_PATH + "columns.avro";
+
+    List<ColumnRow> dbList = new ArrayList<>();
+    List<ColumnRow> avroList = new ArrayList<>();
+
+    try (PreparedStatement preparedStatement =
+        connection.prepareStatement(format(SqlHelper.getSql(sqlPath), DB_NAME, DB_NAME))) {
+      ResultSet rs = preparedStatement.executeQuery();
+
+      while (rs.next()) {
+        ColumnRow dbRow = new ColumnRow();
+        dbRow.dataBaseName = getStringNotNull(rs, "DataBaseName");
+        dbRow.tableName = getStringNotNull(rs, "TableName");
+        dbRow.columnName = getStringNotNull(rs, "ColumnName");
+        dbRow.columnFormat = getStringNotNull(rs, "ColumnFormat");
+        dbRow.columnTitle = getStringNotNull(rs, "ColumnTitle");
+        dbRow.columnLength = getIntNotNull(rs, "ColumnLength");
+        dbRow.columnType = getStringNotNull(rs, "ColumnType");
+        dbRow.defaultValue = getStringNotNull(rs, "DefaultValue");
+        dbRow.columnConstraint = getStringNotNull(rs, "ColumnConstraint");
+        dbRow.constraintCount = getIntNotNull(rs, "ConstraintCount");
+        dbRow.nullable = getStringNotNull(rs, "Nullable");
+        dbList.add(dbRow);
+      }
+    }
+
+    DataFileReader<GenericRecord> dataFileReader = extractAvroDataFromFile(avroFilePath);
+
+    while (dataFileReader.hasNext()) {
+      GenericRecord record = null;
+      record = dataFileReader.next(record);
+
+      ColumnRow avroRow = new ColumnRow();
+      avroRow.dataBaseName = getStringNotNull(record, "DataBaseName");
+      avroRow.tableName = getStringNotNull(record, "TableName");
+      avroRow.columnName = getStringNotNull(record, "ColumnName");
+      avroRow.columnFormat = getStringNotNull(record, "ColumnFormat");
+      avroRow.columnTitle = getStringNotNull(record, "ColumnTitle");
+      avroRow.columnLength = getIntNotNull(record, "ColumnLength");
+      avroRow.columnType = getStringNotNull(record, "ColumnType");
+      avroRow.defaultValue = getStringNotNull(record, "DefaultValue");
+      avroRow.columnConstraint = getStringNotNull(record, "ColumnConstraint");
+      avroRow.constraintCount = getIntNotNull(record, "ConstraintCount");
+      avroRow.nullable = getStringNotNull(record, "Nullable");
+      avroList.add(avroRow);
+    }
+
+    List<ColumnRow> dbListCopy = new ArrayList<>(dbList);
```

Comment:
Could we use `Set`s instead? The removal will be O(n^2). With sets, that will be linear and you could use Guava's set operations.

Reply:
Well, sure, we can do it but think about it from a testing perspective:
If db for some reason has duplicates like `A, B, B` and avro has only `A, B`, so, we don't want to reduce the db list and get `A, B`, doing this will give us zero difference but, obviously, there is a difference.

In addition to that, so far, generating 10 users takes 30-45 sec, and Users test on lists of 200+ elements takes 1-2 secs.
I'm curious to check how long the test takes on a big chunk of data, like 10 000 entries or so.

Reply:
Fair. In that case I would use an implementation of set that preserves order like `LinkedHashSet` or `ImmutableSet`.

Reply:
Done.